feat: Export RegisterObjectKind and MatcherFunc from objectkinds package

heckler1 opened the pull request:
This allows consumers to register arbitrary matchers for arbitrary objectkinds. This enables expanding beyond the builtins, as well as linting custom resources.

janisz commented:
@heckler1 It will be great if you add some docs about how to use it to extend kube-linter beyond builtins

heckler1 commented:
Hey @janisz, good idea! I added a section to the Configuring Kube-Linter document, it seemed like a good place. I also went back and added doc comments to the exported types. Let me know what you think, open to any feedback.

janisz commented on docs/configuring-kubelinter.md:
````diff
@@ -142,3 +148,64 @@ For example,
+
+- Finally, the check template itself
+
+  ```go
````
I wonder if we could add this as [testable example](https://go.dev/blog/examples) to get fast feedback if we broke something and need to update docs.

heckler1 replied:
Cool, T had no idea that was a feature. I've rewritten the doc into a go file with the original code plus a quick test, and changed the markdown to refer to the go file. The example will be tested on `make test`, so we can be notified of any regressions.

janisz commented:
Nice!

janisz commented:
Cool, Thanks!
